### DISCUSSION TITLE
Figure width in typst: should this work?

### DISCUSSION THREAD

**Discussion by andrewpbray** [quarto-dev/quarto-cli#10597] [Q&A] [answered]
### Description

The [docs](https://quarto.org/docs/output-formats/typst.html#known-limitations) state that:

> The default size of images may not reflect the behavior you are used to in other output formats. This is a problem that Typst, pandoc and Quarto are actively working to fix. In the meantime, you can manually [specify image widths](https://quarto.org/docs/authoring/figures.html#figure-sizing).

When you click through the link, you get to [Figure Sizing](https://quarto.org/docs/authoring/figures.html#figure-sizing), which presents this example:

```qmd
---
format: typst
---

![Elephant](https://quarto.org/docs/authoring/images/html-figure.png){width=300}
```

However I'm getting this error when running on 1.5.56 on macos 14.6.1.

```bash
[typst]: Compiling template.typ to template.pdf...error: expected relative length or auto, found integer
    ┌─ template.typ:277:61
    │
277 │ #box(image("template_files/mediabag/html-figure.png", width: 300))
    │    
```
I've tested with local images and different width units but no dice.

The docs make me think this should work?

**Comment by mcanouil** [comment, marked as answer]
I believe this is fixed in 1.6.

For reference:

- #10494

**Reply by andrewpbray** [reply]
Yep, I upgraded to v1.6.6 and it works (with the `pt` units).

Thanks!

**Comment by cscheid** [comment]
Typst requires units on their image sizes, so if you're in an older version of Quarto, use `300pt` instead. (we make this fix for you in 1.6, as @mcanouil alluded)
